### PR TITLE
Made changes to accommodate updating to spring v 4.3

### PR DESCRIPTION
**Pull request by kkoz**

### CODE REVIEW

File: src/main/java/com/glencoesoftware/omero/ms/image/region/ImageRegionVerticle.java [hunk 1]
```diff
@@ -160,6 +147,7 @@ private void renderImageRegion(Message<String> message) {
                 traceCtx);
         span.tag("ctx", message.body());
 
+        log.info(Integer.toString(port));
```
Comment (chris-allan) [inline src/main/java/com/glencoesoftware/omero/ms/image/region/ImageRegionVerticle.java]: Required?

File: src/main/resources/beanRefContext.xml [hunk 2]
```diff
@@ -36,14 +36,43 @@
...
+  <bean id="maxTileLength"
+        class="java.lang.Integer">
+     <constructor-arg value="${omero.pixeldata.max_tile_length:2048}" />
+  </bean>
+
   <bean id="metrics" class="ome.system.metrics.NullMetrics"/>
 
+  <bean id="omero-ms-image-region-verticle"
+        class="com.glencoesoftware.omero.ms.image.region.ImageRegionVerticle">
+    <constructor-arg value="${omero.host:localhost}" />
```
Comment (chris-allan) [inline src/main/resources/beanRefContext.xml]: Does not need to be injected via spring.

File: src/main/resources/beanRefContext.xml [hunk 3]
```diff
@@ -36,14 +36,43 @@
...
+    <constructor-arg type="int" value="${omero.port:4064}" />
+    <constructor-arg ref="/OMERO/Pixels" />
+    <constructor-arg ref="internal-ome.api.ICompress" />
+    <constructor-arg ref="lutProvider" />
+    <constructor-arg ref="maxTileLength" />
+  </bean>
+
+  <bean id="omero-ms-shape-mask-verticle"
+        class="com.glencoesoftware.omero.ms.image.region.ShapeMaskVerticle">
+    <constructor-arg value="${omero.host:localhost}" />
```
Comment (chris-allan) [inline src/main/resources/beanRefContext.xml]: Does not need to be injected via spring.

File: src/dist/conf/config.yaml [hunk 4]
```diff
@@ -23,6 +23,8 @@ omero.server:
     # OMERO_HOME/lib/scripts
     omero.script_repo_root: "/opt/omero/lib/scripts"
     omero.pixeldata.max_tile_length: "2048"
+    omero.host: "localhost" #For verticle bean instantiation
```
Comment (chris-allan) [inline src/dist/conf/config.yaml]: This duplicates this configuration and is not needed as the verticles can pick up this configuration from the Vertx configuration available from `config()`.

File: src/dist/conf/config.yaml [hunk 5]
```diff
@@ -23,6 +23,8 @@ omero.server:
     # OMERO_HOME/lib/scripts
```
Comment (chris-allan) [inline src/dist/conf/config.yaml]: `worker_pool_size` needs to be put in the example configuration. In `master` it has the form:

```
# Size of the worker pool (defaults to CPUs x 2)
#worker_pool_size: 8
```

File: build.gradle [hunk 6]
```diff
@@ -74,14 +74,15 @@ configurations.all {
 dependencies {
     compile 'io.prometheus:simpleclient_vertx:0.6.0'
     compile 'ch.qos.logback:logback-classic:1.2.3'
-    compile ('com.glencoesoftware.omero:omero-ms-core:0.4.2-SNAPSHOT') {
+    compile ('com.glencoesoftware.omero:omero-ms-core:0.4.3-SNAPSHOT') {
```
[the remark this answers is not in the document]
Comment (chris-allan) [reply build.gradle]: Resolved by merging the required upstream PR, glencoesoftware/omero-ms-core#9

File: src/main/java/com/glencoesoftware/omero/ms/image/region/ImageRegionMicroserviceVerticle.java [hunk 7]
```diff
@@ -182,26 +188,32 @@ public void deploy(JsonObject config, Future<Void> future) {
...
                         .setWorker(true)
                         .setMultiThreaded(true)
+                        .setWorkerPoolSize(workerPoolSize)
                         .setConfig(config));
-        vertx.deployVerticle(new ShapeMaskVerticle(
-                omero.getString("host"), omero.getInteger("port")),
+        vertx.deployVerticle("omero:omero-ms-shape-mask-verticle",
                 new DeploymentOptions()
                         .setWorker(true)
                         .setMultiThreaded(true)
```
[the remark this answers is not in the document]
Comment (kkoz) [reply src/main/java/com/glencoesoftware/omero/ms/image/region/ImageRegionMicroserviceVerticle.java]: Yes - will update.

File: src/main/java/com/glencoesoftware/omero/ms/image/region/ImageRegionVerticle.java (hunk 1 above)
Comment (kkoz) [reply src/main/java/com/glencoesoftware/omero/ms/image/region/ImageRegionVerticle.java]: Removing

**Comment by chris-allan** [comment]
@kkoz: For Vertx 3.8.1 we need to update the cookie handling in ms-core, yes?

https://github.com/vert-x3/wiki/wiki/3.8.1-Deprecations-and-breaking-changes#cookie-handling

File: README.md [hunk 8]
```diff
@@ -171,7 +171,7 @@ Eclipse Configuration
         cp src/dist/conf/config.yaml conf/
         # Edit as appropriate
 
-1. Add a new Run Configuration with a main class of `io.vertx.core.Starter`::
+1. Add a new Run Configuration with a main class of `io.vertx.core.Launcher`::
```
Comment (chris-allan) [inline README.md]: `mainClassName` in `build.gradle` also needs to change

**Comment by kkoz** [comment]
w.r.t the `replyAndRequest` changes, I think I made all the necessary ones. I believe the deprecated methods are those which have a `Handler<AsyncResult<Message<T>>>` as an argument. The remaining calls in the project do not have this, and replyAndRequest requires it. I think we are up to date on this point.

**Comment by chris-allan** [comment]
> w.r.t the `replyAndRequest` changes, I think I made all the necessary ones. I believe the deprecated methods are those which have a `Handler<AsyncResult<Message<T>>>` as an argument. The remaining calls in the project do not have this, and replyAndRequest requires it. I think we are up to date on this point.

A ha. That makes sense. 👍

**Pull request merged**